common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -147,10 +149,13 @@ class HTTPClient(
     private fun getHeaders(
         authenticationHeaders: Map<String, String>,
         urlPath: String,
-        refreshETag: Boolean
+        refreshETag: Boolean,
+        gzipRequest: Boolean
     ): Map<String, String> {
+        val contentEncoding = if (gzipRequest) "gzip" else null
         return mapOf(
             "Content-Type" to "application/json",
+            "Content-Encoding" to contentEncoding,
```

Comment:
I don't usually suggest doing things inline but I think for this one it might be a little easier to read the same as the setting for observer mode, i.e.: 
```suggestion
            "Content-Encoding" to if (gzipRequest) "gzip" else null,
```

Reply:
while we're talking about encoding, are we sending accept encoding at all? it'd be a nice optimization (as a separate thing, though)

Reply:
I'm almost certain we are: https://developer.android.com/reference/java/net/HttpURLConnection#performance

<img width="867" alt="image" src="https://user-images.githubusercontent.com/808417/216768746-410737ec-bd81-41a8-92ae-2b3f1a18ce9d.png">

I would have to double-check though.